### PR TITLE
Feature set 01

### PR DESCRIPTION
**Pull request by sharathrnair87**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] Where applicable I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

 1. Support querying PolicySets and PolicyChecks
 2. Returns detailed PolicyCheck output for a given TFC run
 3. `run list` now supports listing ALL runs for that workspace

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [x] Yes
- [ ] No

Retrieving `sentinel` body for a policy check requires v1.42.0 of `hashicorp/go-tfe`

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?
see README.md and attached logs
[policy_check_out.log](https://github.com/AGLEnergyPublic/tfectl/files/14000027/policy_check_out.log)
[policy_set_list.log](https://github.com/AGLEnergyPublic/tfectl/files/14000028/policy_set_list.log)

```bash
tfectl run list --workspace-id ws-nfWASMsu4bMxPp --list-all --status applied --query 'length'
[
  860
]
```

### CODE REVIEW

**Comment by sharathrnair87** [comment]
CodeQL false positive - https://github.com/github/codeql/issues/14159

File: cmd/policy_set.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+			for _, policy := range policySet.Policies {
+				log.Debugf("Processing policies in policySet: %s - %s - %s", policySet.Name, policy.Name, policy.ID)
+				tmpPolicySetPolicyList = append(tmpPolicySetPolicyList, policy.ID)
+			}
+
+			policiesSlice, err := json.Marshal(tmpPolicySetPolicyList)
+			check(err)
+
+			entry := fmt.Sprintf(`{"name":"%s","id":"%s","kind":"%s","global":%v,"workspaces":%v, "workspace_count":%d, "workspace_exclusion":%v, "projects":%v, "project_count":%d, "policies":%v, "policy_count":%d}`, policySet.Name, policySet.ID, policySet.Kind, policySet.Global, string(workspaceSlice), policySet.WorkspaceCount, string(workspaceExclSlice), string(projectSlice), policySet.ProjectCount, string(policiesSlice), policySet.PolicyCount)
```
[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:94]: CodeQL false positive - https://github.com/github/codeql/issues/14159

[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:94]: CodeQL false positive - https://github.com/github/codeql/issues/14159

[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:94]: CodeQL false positive - https://github.com/github/codeql/issues/14159

[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:94]: CodeQL false positive - https://github.com/github/codeql/issues/14159

**Review by A30008279 — APPROVED** [review]
LGTM 🚀

**Pull request merged**